pkg/pod/script.go

```diff
@@ -96,140 +94,162 @@ func convertScripts(shellImageLinux string, shellImageWin string, steps []v1beta
+		c.Command = []string{scriptFile}
+	}
+	c.VolumeMounts = append(c.VolumeMounts, scriptsVolumeMount)
+}
 
-		containers = append(containers, *steps[i].ToK8sContainer())
+// placeDebugScriptInContainers inserts debug scripts into containers. It capsules those scripts to files in intiContainer,
+// then execute those scripts in target containers.
+func placeDebugScriptInContainers(containers []corev1.Container, initContainer *corev1.Container) {
+	for i := 0; i < len(containers); i++ {
+		debugInfoVolumeMount := corev1.VolumeMount{
+			Name:      debugInfoVolumeName,
+			MountPath: filepath.Join(debugInfoDir, fmt.Sprintf("%d", i)),
+		}
+		(&containers[i]).VolumeMounts = append((&containers[i]).VolumeMounts, debugScriptsVolumeMount, debugInfoVolumeMount)
 	}
 
-	// Place debug scripts if breakpoints are enabled
-	if len(breakpoints) > 0 {
-		// If breakpoint is not nil then should add the init container
-		// to write debug script files
-		*placeScripts = true
+	type script struct {
+		name    string
+		content string
+	}
+	debugScripts := []script{{
+		name:    "continue",
+		content: defaultScriptPreamble + fmt.Sprintf(debugContinueScriptTemplate, len(containers), debugInfoDir, RunDir),
+	}, {
+		name:    "fail-continue",
+		content: defaultScriptPreamble + fmt.Sprintf(debugFailScriptTemplate, len(containers), debugInfoDir, RunDir),
+	}}
+
+	// Add debug or breakpoint related scripts to /tekton/debug/scripts
+	// Iterate through the debugScripts and add routine for each of them in the initContainer for their creation
+	for _, debugScript := range debugScripts {
+		tmpFile := filepath.Join(debugScriptsDir, fmt.Sprintf("%s-%s", "debug", debugScript.name))
+		heredoc := names.SimpleNameGenerator.RestrictLengthWithRandomSuffix(fmt.Sprintf("%s-%s-heredoc-randomly-generated", "debug", debugScript.name))
+
+		initContainer.Args[1] += fmt.Sprintf(initScriptDirective, tmpFile, heredoc, debugScript.content, heredoc)
+	}
+}
 
-		type script struct {
-			name    string
-			content string
-		}
-		debugScripts := []script{{
-			name:    "continue",
-			content: defaultScriptPreamble + fmt.Sprintf(debugContinueScriptTemplate, len(steps), debugInfoDir, RunDir),
-		}, {
-			name:    "fail-continue",
-			content: defaultScriptPreamble + fmt.Sprintf(debugFailScriptTemplate, len(steps), debugInfoDir, RunDir),
-		}}
-
-		// Add debug or breakpoint related scripts to /tekton/debug/scripts
-		// Iterate through the debugScripts and add routine for each of them in the initContainer for their creation
-		for _, debugScript := range debugScripts {
-			tmpFile := filepath.Join(debugScriptsDir, fmt.Sprintf("%s-%s", "debug", debugScript.name))
-			heredoc := names.SimpleNameGenerator.RestrictLengthWithRandomSuffix(fmt.Sprintf("%s-%s-heredoc-randomly-generated", "debug", debugScript.name))
-
-			initContainer.Args[1] += fmt.Sprintf(initScriptDirective, tmpFile, heredoc, debugScript.content, heredoc)
+// convertListOfSidecars does the heavy lifting for convertScripts.
```

Comment:
This docstring and `convertListOfSteps` probably could be updated; these descriptions seem to apply more to `placeScriptInContainer`.